### ISSUE TITLE
Invalid Character in biocache-service and bie-index query URLs

### ISSUE THREAD

**Issue by ansell** [AtlasOfLivingAustralia/bie-plugin#122]
A request from bie generated an error in the biocache-service logs due to an invalid character. The offending character is the character encoded as ``\x22`` in the log message below, which is a double-quote character that has not been encoded. Its percent encoded version, which would work, would show in the nginx log files as ``%22``:

```
==> /var/log/tomcat7/catalina.out <==
Oct 26, 2018 1:43:13 PM org.apache.coyote.http11.AbstractHttp11Processor process
INFO: Error parsing HTTP request header
 Note: further occurrences of HTTP header parsing errors will be logged at DEBUG level.
java.lang.IllegalArgumentException: Invalid character found in the request target. The valid characters are defined in RFC 7230 and RFC 3986
	at org.apache.coyote.http11.InternalInputBuffer.parseRequestLine(InternalInputBuffer.java:189)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:992)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
	at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:316)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)


==> /var/log/nginx/access.log <==
52.62.134.156 - - [26/Oct/2018:13:43:13 +1100] "GET /ws/occurrences/search?q=Haematopus+fuliginosus&fq=multimedia:\x22Sound\x22 HTTP/1.1" 400 5 "-" "Java/1.8.0_181" "52.62.134.156" request_time=0.001 upstream_response_time=0.001 upstream_connect_time=0.001 upstream_header_time=0.001 upstream_cache_status=-
```

**Comment by ansell** [comment]
This issue is also happening with queries from bie-plugin to bie-index (/ws/):

```
==> /var/log/tomcat7/catalina.out <==
Oct 31, 2018 10:09:17 AM org.apache.coyote.http11.AbstractHttp11Processor process
INFO: Error parsing HTTP request header
 Note: further occurrences of HTTP header parsing errors will be logged at DEBUG level.
java.lang.IllegalArgumentException: Invalid character found in the request target. The valid characters are defined in RFC 7230 and RFC 3986
	at org.apache.coyote.http11.InternalInputBuffer.parseRequestLine(InternalInputBuffer.java:189)
	at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:992)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
	at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:316)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)


==> /var/log/nginx/access.log <==
127.0.0.1 - - [31/Oct/2018:10:09:17 +1100] "GET /ws/search?q=brushtail%20possum&fq=idxtype:\x22TAXON\x22&start=0&rows=10&sort=&dir=desc&facets=idxtype,rank,locatedInHubCountry,imageAvailable,speciesGroup,taxonomicStatus,conservationStatusAUS_s,conservationStatusACT_s,conservationStatusNSW_s,conservationStatusQLD_s,conservationStatusVIC_s,conservationStatusTAS_s,conservationStatusSA_s,conservationStatusWA_s,conservationStatusNT_s&q.op=OR HTTP/1.1" 400 5 "-" "Java/1.8.0_191" "-"

==> /var/log/tomcat7/catalina.out <==
Failed calling web service. class java.io.IOException Server returned HTTP response code: 400 for URL: https://bie.ala.org.au/ws/search?q=brushtail%20possum&fq=idxtype:"TAXON"&start=0&rows=10&sort=&dir=desc&facets=idxtype,rank,locatedInHubCountry,imageAvailable,speciesGroup,taxonomicStatus,conservationStatusAUS_s,conservationStatusACT_s,conservationStatusNSW_s,conservationStatusQLD_s,conservationStatusVIC_s,conservationStatusTAS_s,conservationStatusSA_s,conservationStatusWA_s,conservationStatusNT_s&q.op=OR URL= https://bie.ala.org.au/ws/search?q=brushtail%20possum&fq=idxtype:"TAXON"&start=0&rows=10&sort=&dir=desc&facets=idxtype,rank,locatedInHubCountry,imageAvailable,speciesGroup,taxonomicStatus,conservationStatusAUS_s,conservationStatusACT_s,conservationStatusNSW_s,conservationStatusQLD_s,conservationStatusVIC_s,conservationStatusTAS_s,conservationStatusSA_s,conservationStatusWA_s,conservationStatusNT_s&q.op=OR.

==> /var/log/tomcat7/ala-bie.log <==
2018-10-31 10:09:17.517 ERROR --- [io-8080-exec-16] g.a.c.au.org.ala.bie.SpeciesController   : Error requesting taxon concept object: Failed calling web service. class java.io.IOException Server returned HTTP response code: 400 for URL: https://bie.ala.org.au/ws/search?q=brushtail%20possum&fq=idxtype:"TAXON"&start=0&rows=10&sort=&dir=desc&facets=idxtype,rank,locatedInHubCountry,imageAvailable,speciesGroup,taxonomicStatus,conservationStatusAUS_s,conservationStatusACT_s,conservationStatusNSW_s,conservationStatusQLD_s,conservationStatusVIC_s,conservationStatusTAS_s,conservationStatusSA_s,conservationStatusWA_s,conservationStatusNT_s&q.op=OR URL= https://bie.ala.org.au/ws/search?q=brushtail%20possum&fq=idxtype:"TAXON"&start=0&rows=10&sort=&dir=desc&facets=idxtype,rank,locatedInHubCountry,imageAvailable,speciesGroup,taxonomicStatus,conservationStatusAUS_s,conservationStatusACT_s,conservationStatusNSW_s,conservationStatusQLD_s,conservationStatusVIC_s,conservationStatusTAS_s,conservationStatusSA_s,conservationStatusWA_s,conservationStatusNT_s&q.op=OR.

==> /var/log/nginx/access.log <==
[REDACTED] - - [31/Oct/2018:10:09:17 +1100] "GET /search?q=brushtail+possum&fq=idxtype:%22TAXON%22 HTTP/1.1" 200 21419 "https://bie.ala.org.au/search?q=brushtail+possum" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36" "-"
```

**Comment by ansell** [comment]
This is affecting every ``/image-search`` and ``/sound-search`` query

**Comment by ansell** [comment]
The /sound-search part of this issue is still broken for biocache-service queries, based on this trace on nectar-bie-test.

```==> /var/log/tomcat7/ala-bie.log <==
2018-11-01 09:51:37.076 ERROR --- [bio-8080-exec-2] g.a.services.au.org.ala.bie.WebService   : {'error': 'Failed to get json from web service. class java.io.IOException Server returned HTTP response code: 400 for URL: https://biocache.ala.org.au/ws/occurrences/search?q=Avena+strigosa&fq=multimedia:"Sound" URL= https://biocache.ala.org.au/ws/occurrences/search?q=Avena+strigosa&fq=multimedia:"Sound".', 'exception': 'java.io.IOException: Server returned HTTP response code: 400 for URL: https://biocache.ala.org.au/ws/occurrences/search?q=Avena+strigosa&fq=multimedia:"Sound"'}

==> /var/log/nginx/access.log <==
[REDACTED] - - [01/Nov/2018:09:51:37 +1100] "GET /sound-search/?s=Avena+strigosa HTTP/2.0" 200 22 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.12; rv:62.0) Gecko/20100101 Firefox/62.0" "-"
```

**Comment by ansell** [comment]
@charvolant fixed more of these issues in #135 so closing this unless they start reappearing again
